src/Fantomas.Core/CodePrinter.fs

```diff
@@ -3472,10 +3472,12 @@ let genTypeDefn (td: TypeDefn) =
                 if isMulti && ctx.Config.AlternativeLongMemberDefinitions then
                     (optSingle genSingleTextNode typeName.EqualsToken) ctx
                 else
-                    (sepSpaceOrIndentAndNlnIfExpressionExceedsPageWidth (
-                        optSingle genSingleTextNode typeName.EqualsToken
-                    ))
-                        ctx)
+                    let sep =
+                        match typeName.EqualsToken with
+                        | Some _ -> sepSpaceOrIndentAndNlnIfExpressionExceedsPageWidth
+                        | None -> sepSpaceOrIndentIfExpressionExceedsPageWidth
```

Comment:
Any ideas on when this can happen (no equal + multiline expression)? The only thing I can think of is a multiline block comment, but that won't be affected by the indentation.

```fsharp
type x (* foo
 bar  *)
```

If it were to come after the type definition:

```fsharp
type x 

(* foo
 bar  *)

let x = 0
```

The trivia assignment will be different altogether.

I think we can just move the `sep` call into the `optSingle`:

```fsharp
optSingle
  (fun eq -> sepSpaceOrIndentAndNlnIfExpressionExceedsPageWidth (genSingleTextNode eq))
  typeName.EqualsToken
  ctx
```

The test still passed for me, and we don't need anything new in `Context.fs`.

Reply:
That's just perfect. Thanks!